### PR TITLE
Example LOB SQL

### PR DESCRIPTION
**Pull request by pkarman**
connects #14353 

### Description
Example SQL.

### CODE REVIEW

**Comment by yoomlam** [comment]
@rajinderbhanot  As requested, here are some ERD diagrams relevant to the example queries.

This command:
```
rake erd filetype=png filename=erd-LOB-overview title="Line of Business" \
  only=RequestIssue,Appeal,HigherLevelReview,SupplementalClaim,Veteran,DecisionIssue \
  attributes=primary_keys,foreign_keys
``` 
generates this overview:
image

---

While this command:
```
rake erd filetype=png filename=erd-LOB-attribs title="Line of Business" \ 
  only=RequestIssue,Appeal,HigherLevelReview,SupplementalClaim,Veteran,DecisionIssue \ 
  attributes=primary_keys,foreign_keys,inheritance,content
```
generates this:
image

[This](http://voormedia.github.io/rails-erd/gallery.html#notations) describes how to interpret the diagram:
> If we model them in a database, they will share a one-to-many relationship. This cardinality is represented with a single arrow.

**Comment by yoomlam** [comment]
@rajinderbhanot To show Veteran connected to Appeal, HLR, and SC; I added `belongs_to :veteran` into those classes.

image

image

File: app/sql/line-of-business.sql [hunk 1]
```diff
@@ -0,0 +1,84 @@
+-- Example SQL for pull Decision Reviews and related data for a specific Line of Business
+
+-- Line of Business is tracked via 'benefit_type' on the 'request_issues' table
+-- In these examples we are using 'vha'
+
+-- @QUERY_NAME: vha_request_issues
```
Comment (yoomlam) [inline app/sql/line-of-business.sql:6]: Added `@QUERY_NAME` to name each query. This is used in the spec file to run individual queries.

File: app/sql/line-of-business.sql [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+-- @QUERY_NAME: vha_appeal_decisions
+select * from decision_issues
+where decision_review_type = 'Appeal'
+  and decision_review_id in (
+    select decision_review_id
+    from request_issues
+    where decision_review_type = 'Appeal'
+      and benefit_type = 'vha'
+);
```
Comment (yoomlam) [inline app/sql/line-of-business.sql:65]: Should this query check for established appeals only (`where established_at is not null`)?

File: spec/sql/line-of-business_spec.rb [hunk 3]
```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+describe "Line of Business extract example", :postgres do
+  include SQLHelpers
+
+  include_context "AMA Tableau SQL"
```
Comment (yoomlam) [inline spec/sql/line-of-business_spec.rb:6]: Included for background noise. We can remove if desired.

File: app/sql/line-of-business.sql (hunk 2 above)
Comment (pkarman) [reply app/sql/line-of-business.sql:65]: it wouldn't hurt. there's an implicit assumption that an appeal that has not been established would never have any decisions.

File: spec/sql/line-of-business_spec.rb [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  end
+
+  it "compiles SQL" do
+    sql_statements = read_sql("line-of-business").split(";")
+    sql_statements.each do |sql|
+      result = ApplicationRecord.connection.exec_query(sql)
+      expect(result.to_ary).to be_a(Array)
+    end
+  end
+
```
Comment (pkarman) [inline spec/sql/line-of-business_spec.rb:39]: maybe an explicit test for `read_sql_as_hash` to make sure we are parsing correctly?

Comment (yoomlam) [reply spec/sql/line-of-business_spec.rb:39]: Done.

**Pull request merged**